CppCoreGuidelines.md

```diff
@@ -19296,26 +19296,38 @@ The [standard](http://eel.is/c++draft/cpp.include) provides flexibility for comp
 the two forms of `#include` selected using the angle (`<>`) or quoted (`""`) syntax. Vendors take
 advantage of this and use different search algorithms and methods for specifying the include path.
 
-Nevertheless, the guidance is to use the quoted form for including files that exist at a relative path to the file containing the `#include` statement (from within the same component or project) and to use the angle bracket form everywhere else, where possible. This encourages being clear about the locality of the file relative to files that include it, or scenarios where the different search algorithm is required. It makes it easy to understand at a glance whether a header is being included from a local relative file versus a standard library header or a header from the alternate search path (e.g. a header from another library or a common set of includes).
+Nevertheless, the guidance is to use the quoted form for including files that exist at a relative path
+to the file containing the `#include` statement and to use
+the angle bracket form everywhere else, where possible. This encourages being clear about the locality
+of the file relative to files that include it, or scenarios where the different search algorithm is
+required. It makes it easy to understand at a glance whether a header is being included from a locally
+relative file, versus a standard library header or a header from the header search path (e.g. a header
+from another library or from a common set of includes).
```

Comment:
(Moving my reply inline, so that we can thread replies if needed)

> The example tries to draw a distinction between 3 types of include locations: locally relative (edit: just "relative"); standard library; another library, common includes.
>
> We know 1 is the ""-form, and 2 is the <>-form, but we are struggling to find a name for 3, which verbosely I would call "the angle-bracket search path configurable by the user". If we refer to it as "header search path" it's ambiguous whether this is the "" search or the <> search path (and this matters, because the guideline is attempting to espouse that headers in 3 be located from the <> search path)

Well, that's the tautological part. If the guideline intends to say merely "You should use `""` for headers you intend to find via the `""` search path, and `<>` for headers you intend to find via the `<>` search path," then that's completely tautological. I think the useful guideline in this area would say something about which kinds of headers you _should_ try to find via each of those search paths. E.g., I'm sure we agree "You should always intend to find `iostream` via the `<>` search path."

Btw, I agree that "locally relative" is a bit unclear and/or tautological, but consider that `#include <sys/types.h>` provides a pathname relative to _somewhere_. Essentially nobody ever uses `#include` on an _absolute_ path. So the (ill-defined) "local" part of the phrase is actually much more important to the meaning than the "relative" part. (Unfortunately when we try to pin it down we end up trying to define "project" or "codebase" again, which I want to push out of scope for this PR if at all humanly possible.)

Reply:
> If the guideline intends to say merely "You should use "" for headers you intend to find via the "" search path, and <> for headers you intend to find via the <> search path," 

I think the intent is clear from the title, verbatim:
> "Prefer the quoted form of #include for files relative to the including file and the angle bracket form everywhere else"

"" is for relative files (not "" files) and <> is for everything else (not <> files)
That's important because there can be _non_-relative files in "" (e.g. Apple's user-header-search-path) and there can be _relative_ files in <> (e.g. codebases like the one you mention that exclusively use <>). 
The guideline attempts to give examples of files that should be included using each type of search, and the phrase "a header from the header search path" is currently ambiguous in the example because it is important to clarify for the listed examples "another or common libraries" _which_  header search path should be used (e.g. the <>-form)

> #include <sys/types.h> provides a pathname relative to somewhere

and a path is _local_  to somewhere too; "relative" and "local" are terms that require context. The implied context is "to the including file", and this is explicitly given in the title and at least 3 more times in the text of the guideline. I think using "relative" from thereon is reasonable to imply it is "relative to the including file", but if that's not clear then I think it's also fine to simply say "relative to the including file" anytime "relative" is mentioned (a bit verbose, but if there's confusion and it helps to dispel it, then it's worth it).


I found another usage of "alternate search path":
> the different search algorithm is required

We should probably change any occurrence of "different" or "alternate" search path to a term we come up with for the <>-search that is configurable by the user. Or we don't want to term it, then another option is to rephrase it to just 2 distinctions "relative" and "non-relative" (but this also isn't specific to this PR, so it only needs to be addressed if we're also addressing usage of "alternate").